proposals/xxxx-package-manager-branch-support.md

```diff
@@ -0,0 +1,48 @@
+# Package Manager Support for branches in the package manifest
+
+* Proposal: [SE-XXXX](xxxx-package-manager-branch-support.md)
+* Author: [Boris Bügling](https://github.com/neonichu)
+* Review Manager: TBD
+* Status: Discussion
+
+## Introduction
+
+This proposal adds enhancements to the package manifest to support development of packages without strict versioning. It is related to "top of tree" development, but described in a separate proposal, because the implementation is entirely orthogonal to it.
+
+## Motivation
+
+The package manager currently supports packages depedencies which are strictly versioned according to semantic versioning. This works well for users of packages, but we still see some use cases where this hinders common development workflows:
+
+- bootstrapping a new package which does not yet have a version at all
+- allowing references to packages in branches instead of tags -- this is useful for working on revlocked packages
+
+## Proposed solution
+
+As a solution to this problem, we propose to extend the package manifest to allow specifying a branch instead of a version to support revlocked packages and initial bootstrapping.
+
+## Detailed Design
+
+We will introduce a new initializer for `Package` which takes a branch instead of a version range:
```

Comment:
Just to make it clear that this isn't replacing the prior initializer, I would call this a "second" initializer, not a "new" initializer. Also, this should be reworded slightly to make it clear that it's an initializer for the .Package dependency enum, not for the top level Package object.